site-src/geps/gep-713.md

```diff
@@ -0,0 +1,496 @@
+precedence over the default drainTimeout value attached to the Route. At the
+same time, we can see that the default connectionTimeout attached to the Route
+has precedence over the default attached to the Gateway.
+
+![Hierarchical Policy Example](images/713-policy-hierarchy.png)
+
+#### Supported Resources
+It is important to note that not every implementation will be able to support
+policy attachment to each resource described in the hierarchy above. When that
+is the case, implementations should clearly document which resources a policy
+may be attached to.
+
+#### Attaching Policy to GatewayClass
+GatewayClass may be the trickiest resource to attach policy to. Policy
+attachment relies on the policy being defined within the same scope as the
+target. This ensures that only users with write access to a policy resource in a
+given scope will be able to modify policy at that level. Since GatewayClass is a
+cluster scoped resource, this means that any policy attached to it must also be
+cluster scoped.
+
+GatewayClass parameters provide an alternative to policy attachment that may be
+easier for some implementations to support. These parameters can similarly be
+used to set defaults and requirements for an entire GatewayClass.
+
+### Targeting External Services
+In some cases (likely limited to mesh) we may want to apply policies to requests
+to external services. To accomplish this, implementations can choose to support
+a refernce to a virtual resource type:
+
+```yaml
+kind: RetryPolicy
+apiVersion: networking.acme.io/v1alpha1
+metadata:
+  name: foo
+spec:
+  default:
+    maxRetries: 5
+  targetRef:
+    group: networking.acme.io
+    kind: ExternalService
+    name: foo.com
+```
+
+### Apply Policies to Specific Matches
+In some cases we may want to target specific matches within nested objects. For
+example, it may be useful to attach policies to a specific Gateway listener or
+Route rule.
+
+Each Route rule or Gateway listener should be expanded with an optional name
+field. The target ref would be expanded with an optional sectionName field that
+could be used to refer to that specific section of the resource. It would refer
+to the following concepts on these resources:
+
+* Gateway.Listeners.Name
+* xRoute.Rules.Name
+* Service.Ports.Name
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-app-1
+  labels:
+    app: foo
+spec:
+  hostnames:
+  - "foo.com"
+  rules:
+  - name: bar
+    matches:
+    - path:
+        type: Prefix
+        value: /bar
+    forwardTo:
+    - serviceName: my-service1
+      port: 8080
+---
+kind: RetryPolicy
+apiVersion: networking.acme.io/v1alpha2
+metadata:
+  name: foo
+spec:
+  maxRetries: 5
+  targetRef:
+    name: foo
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    sectionName: bar
+```
+
+This would require adding a `Name` field to Gateway listeners and Route rules:
```

Comment:
I think this is a good idea regardless.

Reply:
Agreed, I think this will end up being useful for more things in the future.